src/elements/content-explorer/ContentExplorer.js

```diff
@@ -725,6 +729,38 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    /**
+     * Returns an object with newCollection and newSelected properties.
+     * newCollection will be the given collection, with items.selected properties
+     * updated according to the given selected param. newSelected will be the selected
+     * item if it is in newCollection, otherwise undefined.
+     *
+     * @private
+     * @param {Collection} collection - collection that needs to be updated
+     * @param {?BoxItem} selected - item that should be selected in that collection (if present)
+     * @return {Object}
+     */
+    updateCollection(collection: Collection, selected: ?BoxItem): Object {
+        const newCollection: Collection = { ...collection };
+        const targetID = selected ? selected.id : null;
+
+        const ret = { newCollection, newSelected: undefined };
```

Comment:
It may be more clear to build this object at the time when it's returned:

```
return {
   newCollection,
   newSelected: undefined,
};
```

It also seems a little odd that `updateCollection` returns a wrapper around the collection rather than the collection itself.